### PR TITLE
Fix sourcemaps with kernel

### PR DESCRIPTION
**Pull request by jakemac53**
Please forgive all the commits here - I couldn't easily do a clean rebase so I just did a merge - will squash when merging.

Fixes https://github.com/dart-lang/sdk/issues/36236

- Fixes the inline source maps to strip the schemes from the sources
- Removes a ddc specific hack that was appending `.ddc` to module names before looking up the source map - this breaks kernel. (will remove entirely in a followup that drops ddc support)
- Cleaned up how the stack trace mapper app is built, following the same strategy as the injected client in webdev (you can now run `pub run build_runner watch web` in `build_web_compilers` and get a decent dev cycle).

### CODE REVIEW

File: build_web_compilers/build.yaml [hunk 1]
```diff
@@ -68,6 +81,15 @@ builders:
         compiler: dart2js
     applies_builders:
       - build_web_compilers|dart2js_archive_extractor
+  _stack_trace_mapper_copy:
```
Comment (natebosch) [inline build_web_compilers/build.yaml]: This is going to pollute generated build scripts for everyone. Not a huge impact but I wonder if we should avoid it?

Should we make "private" builders a first class concept?

Should we create a separate package for this in the mono_repo which is a `dev_dependency` instead?

File: build_web_compilers/tool/copy_builder.dart [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import 'package:build/build.dart';
+
+/// Factory for the build script.
+Builder copyBuilder(_) => _CopyBuilder();
+
+/// Copies the [_stackTraceMapperJs] file to [_stackTraceMapperCopyJs].
+class _CopyBuilder extends Builder {
+  @override
+  Map<String, List<String>> get buildExtensions => {
```
Comment (natebosch) [inline build_web_compilers/tool/copy_builder.dart]: [optional] 

```dart
@override
final buildExtensions = {
  _stackTraceMapperJs.path: [_stackTraceMapperCopyJs.path],
};
```

File: build_web_compilers/build.yaml (hunk 1 above)
Comment (jakemac53) [reply build_web_compilers/build.yaml]: It doesn't actually pollute them - the builder is not even importable outside this project.

Comment (natebosch) [reply build_web_compilers/build.yaml]: Ah I forgot about that setup.

**Pull request merged**